deps/rust_store/src/lib.rs

```diff
@@ -168,129 +188,85 @@ pub async fn connect_and_test(credentials: &AzureCredentials) -> anyhow::Result<
-                    tracing::trace!("Instant BW = {:.3} MiB/s", ((bytes - last_bytes) as f64 / 1024.0 / 1024.0));
-                    last_bytes = bytes;
-                }
-            });
-        }
-
         rx.map(|req| {
             async {
                 match req {
                     Request::Get(p, slice, credentials, response, notifier) => {
-                        let client = match CLIENTS.try_get_with(credentials.get_hash(), connect_and_test(credentials)).await {
+                        let client = match CLIENTS.try_get_with(credentials.get_hash(), connect(credentials)).await {
                             Ok(client) => client,
                             Err(e) => {
                                 response.from_error(e);
                                 notifier.notify();
                                 return;
                             }
                         };
-                        let mut tries = 10;
-                        loop {
-                            match client.get(&p).await {
-                                Ok(result) => {
-                                    let chunks = result.into_stream().collect::<Vec<_>>().await;
-                                    if let Some(Err(e)) = chunks.iter().find(|result| result.is_err()) {
-                                        if tries > 0 {
-                                            tracing::trace!("error while fetching a chunk, retrying: {}", e);
-                                            tries -= 1;
-                                            continue;
-                                        } else {
-                                            tracing::debug!("{}", e);
-                                            response.from_error(e);
-                                            notifier.notify();
+                        match client.get(&p).await {
+                            Ok(result) => {
+                                let chunks = result.into_stream().collect::<Vec<_>>().await;
+                                if let Some(Err(e)) = chunks.iter().find(|result| result.is_err()) {
+                                        tracing::warn!("{}", e);
+                                        response.from_error(e);
+                                        notifier.notify();
+                                        return;
+                                }
+                                tokio::spawn(async move {
+                                    let mut received_bytes = 0;
+                                    let mut failed = false;
+                                    for chunk in chunks {
+                                        let chunk = match chunk {
+                                            Ok(c) => c,
+                                            Err(_e) => {
+                                                unreachable!("checked for errors before");
```

Comment:
i know this bit of code didn't change in this PR, but whilst we're tidying things up...

it took me a while to work out why `chunk` cannot be an `Err` here 

it's because we already checked above (in `chunks.iter().find(|result| result.is_err())`)

I wonder if the code could more easily show us this connection, e.g. by using the same variable name (above we use `result` whereas here we use `chunk`), and making the `unreachable!` message a little more specific, e.g.
```suggestion
                                    for result in chunks {
                                        let chunk = match result {
                                            Ok(c) => c,
                                            Err(_e) => {
                                                unreachable!("checked `chunks` for errors before calling `spawn`");
```